Slow query response times

### Issue 
A simple query using the `up{}` metric for the last 7 days or even just the last 3-4 results in a 3-5 second delay. The same query going directly to either of the VictoriaMetrics backends results in only a 130-170ms delay. I understand that Promxy has to query all the servers in the server group and then merge the results but a ~180% increase in query times seems a little much for my simple setup.  

### Setup
So I have a very simple setup of two instances. Each instance is running VictoriaMetrics (version 1.20.7) and Prometheus (2.9.2). I have one of those instances running Promxy (v0.0.41) with the two VictoriaMetrics instances in a server group. 

Promxy config:
```
global:
  evaluation_interval: 5s
  external_labels:
    source: promxy

promxy:
  server_groups:
    - static_configs:
      - targets:
        - prometheus1-a:8428
        - prometheus1-b:8428
      labels:
        sg: VM_Cluster1
      anti_affinity: 10s
      remiote_read: false
      http_client:
        diial_timeout: 2s
        tls_config:
          insecure_skip_verify: true
```

### Results
Going directly to a VictoriaMetric instance
![Screen Shot 2019-07-02 at 11 56 28 AM](https://user-images.githubusercontent.com/2513533/60528496-8c1fe600-9cc2-11e9-868c-1a067df9ef95.png)

![Screen Shot 2019-07-02 at 11 56 38 AM](https://user-images.githubusercontent.com/2513533/60528495-8c1fe600-9cc2-11e9-9a0b-0f37b9696641.png)

Going directly to Promxy
![Screen Shot 2019-07-02 at 11 54 47 AM](https://user-images.githubusercontent.com/2513533/60528683-f6d12180-9cc2-11e9-9d5a-66738375b7e0.png)

![Screen Shot 2019-07-02 at 11 55 01 AM](https://user-images.githubusercontent.com/2513533/60528682-f6d12180-9cc2-11e9-977b-d3e7ba419c2d.png)

Here's the actual Promxy log line entries when doing the following queries:
**Last 2 days:** `[02/Jul/2019 16:22:00] "GET /api/v1/query_range HTTP/1.1 200 6945" 1.794222 start=1561911720&end=1562084520&step=120&query=up%7B%7D`
**Last 7 days:**`[02/Jul/2019 16:23:07] "GET /api/v1/query_range HTTP/1.1 200 5163" 3.456831 start=1561479600&end=1562084400&step=300&query=up%7B%7D`

Let me know if you need any other additional info. Thanks

@jacksontj I tried doing a `make release` with the master branch but get the following errors:
```
# github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/scrape
/root/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/scrape/scrape.go:276:11: sp.client.CloseIdleConnections undefined (type *http.Client has no field or method CloseIdleConnections)
/root/go/src/github.com/jacksontj/promxy/vendor/github.com/prometheus/prometheus/scrape/scrape.go:334:11: oldClient.CloseIdleConnections undefined (type *http.Client has no field or method CloseIdleConnections)
An error has occurred! Aborting the script execution...
make: *** [release] Error 1
```

--- a/pkg/proxystorage/proxy.go
+++ b/pkg/proxystorage/proxy.go
@@ -388,11 +388,35 @@ func (p *ProxyStorage) NodeReplacer(ctx context.Context, s *promql.EvalStmt, nod
 
 	// If we are simply fetching a Vector then we can fetch the data using the same step that
 	// the query came in as (reducing the amount of data we need to fetch)
-	// If we are simply fetching data, we skip here to let it fall through to the normal
-	// storage API
 	case *promql.VectorSelector:
-		// Do Nothing
-		return nil, nil
+		// If the vector selector already has the data we can skip
+		if n.HasSeries() {
+			return nil, nil
+		}
+
+		var result model.Value
+		var warnings api.Warnings
+		var err error
+		if s.Interval > 0 {
+			result, warnings, err = state.client.QueryRange(ctx, n.String(), v1.Range{
+				Start: s.Start.Add(-offset - promql.LookbackDelta),
+				End:   s.End.Add(-offset),
+				Step:  s.Interval,
+			})
+		} else {
+			result, warnings, err = state.client.Query(ctx, n.String(), s.Start.Add(-offset))
+		}
+
+		if err != nil {
+			return nil, errors.Cause(err)
+		}
+
+		iterators := promclient.IteratorsForValue(result)
+		series := make([]storage.Series, len(iterators))
+		for i, iterator := range iterators {
+			series[i] = &proxyquerier.Series{iterator}
+		}
+		n.SetSeries(series, promhttputil.WarningsConvert(warnings))
 
 	// If we hit this someone is asking for a matrix directly, if so then we don't
 	// have anyway to ask for less-- since this is exactly what they are asking for

